### ISSUE TITLE
`make_pairwise_predict` truncates the dataset to an even length, leading to a stability of 2 for one inserted element

### ISSUE THREAD

**Issue by TedTed** [opendp/opendp#2950]
This problem is visible in the [tutorial](https://docs.opendp.org/en/stable/api/user-guide/plugins/theil-sen-regression.html): the first line of `pairwise_predict` removes the last element of the table if it's odd. So adding a new element to an even-length table removes another element, leading to a stability of 2; the problem also exists in the actual codebase. The easiest way to see this is to run the private method directly, and see that its output changes entirely (symmetric distance 2) when adding a single element to the input (symmetric distance 1):

```py
import numpy as np
import opendp.prelude as dp
from opendp.extras.sklearn.linear_model._make_private_theil_sen import make_pairwise_predict
dp.enable_features("contrib", "honest-but-curious", "idealized-numerics")

predict = make_pairwise_predict(np.array([25.0, 75.0]), runs=1)
x = np.array([[1.0, 1.0], [2.0, 3.0]])
y = np.array([[9.0, 3.0], [1.0, 1.0], [2.0, 3.0]])  # one added row: symmetric distance 1

print("claimed stability:", predict.map(1))
print("predictions from x:", predict(x).tolist())
print("predictions from y:", predict(y).tolist())
```

It's also possible to observe the privacy leakage empirically. The following repro puts 20 points in the x=y line (so every pair predicts 25 and 75 at the two cuts), and adds (25,0) at the beginning, which predicts 0 at x=25 and 100+ at x=75, pulling medians away. Of course, due to the post-processing median step, the empirical leakage is not as bad as 2ε, more like 1.2ε; this is still not great (and can probably be optimized if we really wanted to).

```py
import math
import numpy as np
import opendp.prelude as dp
dp.enable_features("contrib", "honest-but-curious")

line = [[float(t), float(t)] for t in range(26, 46)]
x = np.array(line)
y = np.array([[25.0, 0.0]] + line)  # one added record

model = dp.sklearn.linear_model.LinearRegression(
    dp.max_divergence(), x_bounds=[(0, 100)], y_bounds=(0, 100), scale=1.0, candidates_count=101
)
def pulled(data):
    slope, intercept = model.measurement(data)
    return slope * 25 + intercept < 24.5 and slope * 75 + intercept > 75.5

runs = 20_000
pulled_x = sum(pulled(x) for _ in range(runs))
pulled_y = sum(pulled(y) for _ in range(runs))
print("claimed epsilon: ", model.measurement.map(1))
print("measured epsilon:", math.log(pulled_y / pulled_x))

```

The simple fix is to multiply the stability of this transformation by 2, with the caveat that the shuffle step in the same transformation has [not really well-defined sensitivity](https://github.com/opendp/opendp/issues/2928) right now.
